libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -881,6 +930,70 @@ string YulUtilFunctions::readFromStorageDynamic(Type const& _type, bool _splitFu
 	});
 }
 
+string YulUtilFunctions::readFromMemory(Type const& _type, bool _fromCalldata, bool _padToWords)
```

Comment:
```suggestion
string YulUtilFunctions::readFromMemoryOrCalldata(Type const& _type, bool _fromCalldata, bool _padToWords)
```

or maybe have two separate functions?

Reply:
Or at least have to separate functions in `YulUtilFunctions` that then call this function with the proper bool value?

Reply:
Also padding is only relevant for byte array elements, and I think it would be better to have a separate function for them.